generation/main.go

```diff
@@ -8,12 +8,14 @@ import (
 	"strconv"
 )
 
-// GenFunc represent a translation between raw tuple (r,g,b) into something else
-type GenFunc func(r, g, b int) ([]string, error)
+// GenFunc represent a translation between raw tuple (r,g,b,index) into something else
+// Index provides the current CSV row so that we can run specific actions based on it.
```

Comment:
Can we abstract fact it is using a `CSV` file ? I would better use something like this : 

> Index provide the index of the current color palette being converted. Beware that index will be reset to 0 once switching to another color palette.

Reply:
Good idea. I've updated this now.